src/utils/generatePermalink.js

```diff
@@ -0,0 +1,27 @@
+"use strict";
+
+const config = require("../_data/config.json");
+const getLang = require("./getLang.js");
+const slugFilter = require("../../node_modules/eleventy-plugin-fluid/src/filters/slug-filter.js");
+const translations = require("../_data/translations.json");
+
+module.exports = (data, collectionType) => {
+    /* If this post is a "stub" with no localized title, we assume it does not exist and prevent it from building. */
+    if (!Object.prototype.hasOwnProperty.call(data, "title")) {
+        return false;
+    }
+
+    const lang = getLang(data.page.filePathStem, collectionType);
+    const shortLang = lang.split("-")[0];
+    const slug = slugFilter(data.title);
+
+    if (collectionType === "pages") {
+        if (data.page.fileSlug === lang) {
+            return (lang === config.languages[0].code) ? "/" : `/${shortLang}/`;
+        }
+
+        return (lang === config.languages[0].code) ? `/${slug}/` : `/${shortLang}/${slug}/`;
+    } else {
+        return (lang === config.languages[0].code) ? `/${translations[lang][collectionType]}/${slug}/` : `/${shortLang}/${translations[lang][collectionType]}/${slug}/`;
+    }
+};
```

Comment:
What would happen here if you have translations for multiple locales? e.g. en-CA, en-UK, en-US? I'd suppose that if en-CA is the default that you'd get that from the default URL. However, the en-UK and en-US would probably override/merge for the /en/ paths.

Reply:
@jobara True— I guess my expectation is that as a starter kit this is a good approach that can be modified for more complex cases where a bunch of national dialects of the same language are being used. In that instance someone could use the full code in the permalink instead of the `shortLang` with minimal changes. The goal here was to make the URL as simple as possible (`/fr/articles` instead of `/fr-CA/articles`) but still be able to add the HTML `lang` attribute in a way that reflects the full culture code (language plus country).

Reply:
An alternative would be to check if there are multiples of the same language with just different locales. In that way you could use the short-code if the language is unique and the full locale if not. This would be more complex in the code though.

In general I guess you need to balance the simplicity of the code, the url, and potential user issues. If you leave as is, it should at least be documented, as I'm not sure how obvious it would be. In particular, if the language can be added via the CMS directly, I'm not sure that the maintainer would be able to debug this issue.

Reply:
> An alternative would be to check if there are multiples of the same language with just different locales. In that way you could use the short-code if the language is unique and the full locale if not. This would be more complex in the code though.

I've implemented this in 53903f5.